tests/ops/functions/test_iterative_qpe.py

```diff
@@ -237,3 +237,51 @@ def circuit_iterative():
             return [qml.expval(op=i) for i in measurements]
 
         assert np.allclose(circuit_qpe(), circuit_iterative())
+
+
+@pytest.mark.slow
+@pytest.mark.jax
+def test_capture_execution(seed):
+    """Test that iterative qpe can be captured and executed.
+
+    While this is a rather bad test:
+    * the captured jaxpr has too many classical instructions for
+    easy verification of its contents
+    * The captured jaxpr cannot be used with CollectOpsandMeas as it converts mcm integers to
+    measurement values, which are incompatible with the scatter operation used in
+    `measurements = measurements.at[iters - i - 1].set(m)`
+    * Evaluating jaxpr currently uses single-branch-statistics, which gives incorrect results for a
+    a single execution.
+
+
+    """
+
+    qml.capture.enable()
+    import jax
+
+    def f(x):
+        qml.X(0)
+        return qml.iterative_qpe(qml.RZ(x, wires=[0]), aux_wire=1, iters=3)
+
+    x = jax.numpy.array(2.0)
+
+    jaxpr = jax.make_jaxpr(f)(1.5)
+
+    dev = qml.device("default.qubit", wires=3, seed=seed)
+
+    # hack for single branch statistics
+    samples = qml.math.vstack([dev.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts, x) for _ in range(5000)])
```

Comment:
Oh man, hacky dynamic one shot 😬 

Unfortunately, I have absolutely no ideas about how else to test this other than verifying that the jaxpr equations are correct, which is not feasible.